pycbc/io/hdf.py

```diff
@@ -476,68 +493,69 @@ def mask_to_n_loudest_clustered_events(self, n_loudest=10,
                 new_times.append(curr_time)
             if len(new_index) >= n_loudest:
                 break
+
         index = np.array(new_index)
+        index.sort()
```

Comment:
I think adding this added a bunch of complication in the exe. Why is it needed? With only 10 - 100 points anything using this is not slow.

Reply:
I'm not sure what you are commenting on here.

Reply:
The only real changes to this functions are to properly pick up the possible mask types and to do the masking *before* trying to calculate all the statistics. Is that latter thing what you are commenting on? (line 467) 

That makes a huge difference actually, as you were *not* calculating the statistic for 10-100 points, you were calculating it for *all* points and then selecting a small number of the result. That was a *huge* part of the problem. The full size (before making) can be O(10^8-9) and since the statistic may need to read many fields, it was blowing up the memory useage to 50 - 100 GB depending on chunk I was looking at.